tensorrt_llm/_torch/pyexecutor/llm_request.py

```diff
@@ -35,10 +35,199 @@
 }
 
 
+class LogitsStorage:
+    beam_width: int = -1
+    vocab_size: int
+
+    def __init__(self, seq_length: int, device_memory=True):
```

Comment:
How about `use_device_memory` or something else? `device_memory` as a bool is a bit odd.